### PR TITLE
[NoSquash] Replace settings tab with button

### PR DESCRIPTION
**Pull request by rubenwardy**
Part of #13476

image
image
image

## To do

This PR is Ready for Review

- [x] Icon

## How to test

* Open main menu
* Click settings button
* See settings

### CODE REVIEW

**Comment by Zughy** [comment]
Graphically speaking is pretty bad, the i con was more subtle. Icon + button, if any

**Comment by rubenwardy** [comment]
This is still a draft PR - I was going to ask you for the icon again

**Comment by CallMeDax** [comment]
I think an icon would be much better than a button, but the problem is that the art style wouldn't match the rest of the main menu. imo the tabs and other buttons should also be changed to have more of a pixel art style. possibly for a different PR. Also I think an icon is enough for the settings, since less text means less translations overflowing to the side.

**Comment by rubenwardy** [comment]
Updated with icon provided by Zughy

**Review by srifqi — CHANGES_REQUESTED** [review]
Other than the bug on Windows, the PR looks good to me.

File: builtin/fstk/tabview.lua [hunk 1]
```diff
@@ -76,7 +76,25 @@ local function get_formspec(self)
 		end
 	end
 
-	local formspec = (prepend or "") .. self:tab_header() .. content
+	local end_button_size = 0.65
```
Comment (grorp) [inline builtin/fstk/tabview.lua:79]:
```suggestion
	local end_button_size = 0.75
```

The button is hard to hit on Android, I suggest increasing the size to `0.75`.

File: builtin/fstk/tabview.lua [hunk 2]
```diff
@@ -76,7 +76,25 @@ local function get_formspec(self)
...
+	end
+
+	local formspec = (prepend or "") .. self:tab_header(tab_header_size) .. content
+
+	if self.end_button then
+		formspec = formspec ..
+				("style[%s;noclip=true;border=false]"):format(self.end_button.name) ..
+				("tooltip[%s;%s]"):format(self.end_button.name, self.end_button.label) ..
+				("image_button[%f,%f;%f,%f;%s;%s;]"):format(
+						self.width - end_button_size, -0.43 - end_button_size / 2,
```
Comment (grorp) [inline builtin/fstk/tabview.lua:93]:
```suggestion
						self.width - end_button_size,
						-tab_header_size.height / 2 - end_button_size / 2,
```

**Review by grorp — CHANGES_REQUESTED** [review]

**Review by srifqi — APPROVED** [review]
Except a small change to the documentation, this PR looks good to me.

**Review by grorp — APPROVED** [review]

**Pull request merged**